stdlib/websub/src/main/java/org/ballerinalang/net/websub/BallerinaWebSubConnectorListener.java

```diff
@@ -271,12 +270,10 @@ private ObjectValue createNotification(ObjectValue httpRequest) {
     private MapValue createCustomNotification(HttpCarbonMessage inboundRequest, AttachedFunction resource,
                                               ObjectValue httpRequest) {
         BRecordType recordType = webSubServicesRegistry.getResourceDetails().get(resource.getName());
-        MapValue<String, Object> customNotification =
-                BallerinaValues.createRecordValue(recordType.getPackage().getName(), recordType.getName());
         MapValue<String, ?> jsonBody = getJsonBody(httpRequest);
         inboundRequest.setProperty(ENTITY_ACCESSED_REQUEST, httpRequest);
         if (jsonBody != null) {
-            return JSONUtils.convertJSONToRecord(jsonBody, (BStructureType) customNotification.getType());
+            return JSONUtils.convertJSONToRecord(jsonBody, recordType);
```

Comment:
If the field does not exists in the JSON, set the default value for that struct field via this method. Is it fine? Still a record is returned with default values for a totally different JSON.

Reply:
Changed with 0296071 to use constructFrom instead. Thanks!